File: sdks/thunkmetrc-java/wrapper/src/main/java/io/github/thunkier/thunkmetrc/wrapper/services/FacilitiesService.java
package io.github.thunkier.thunkmetrc.wrapper.services;

import io.github.thunkier.thunkmetrc.client.MetrcClient;
import io.github.thunkier.thunkmetrc.wrapper.MetrcRateLimiter;import io.github.thunkier.thunkmetrc.wrapper.models.*;import java.util.List;

public class FacilitiesService {
    private final MetrcClient client;
    private final MetrcRateLimiter rateLimiter;

    public FacilitiesService(MetrcClient client, MetrcRateLimiter rateLimiter) {
        this.client = client;
        this.rateLimiter = rateLimiter;
    }

    /**
     * This endpoint provides a list of facilities for which the authenticated user has access.
     * GET /facilities/v2
     * @throws Exception If request fails
     * @return Response object
     */
    @SuppressWarnings("unchecked")
    public List<Facility> getFacilities(
        
    ) throws Exception {
        return rateLimiter.execute(null,true,
            () -> (List<Facility>) client.facilities.getFacilities(
                
            )
        );
    }

}

File: sdks/thunkmetrc-kotlin/wrapper/src/main/kotlin/io/github/thunkier/thunkmetrc/wrapper/services/FacilitiesService.kt
package io.github.thunkier.thunkmetrc.wrapper.services

import io.github.thunkier.thunkmetrc.client.MetrcClient
import io.github.thunkier.thunkmetrc.wrapper.MetrcRateLimiter
import io.github.thunkier.thunkmetrc.wrapper.MetrcExtensions
import io.github.thunkier.thunkmetrc.wrapper.models.*
import kotlinx.coroutines.flow.Flow

class FacilitiesService(private val client: MetrcClient, private val rateLimiter: MetrcRateLimiter) {

    /**
     * This endpoint provides a list of facilities for which the authenticated user has access.
     * GET /facilities/v2
     * @throws Exception If request fails
     * @return Response object
     */
    suspend fun getFacilities(): List<Facility>? {
        return rateLimiter.execute(null,true,
        ) { 
            client.facilities.getFacilities() 
        } as? List<Facility>
    }
}

